scipy/stats/_distn_infrastructure.py

```diff
@@ -618,21 +618,24 @@ def random_state(self):
     def random_state(self, seed):
         self._random_state = check_random_state(seed)
 
-    def __getstate__(self):
-        return self._updated_ctor_param(), self._random_state
+    def _attach_argparser_methods(self):
+        """
+        Generates the argument-parsing functions dynamically and attaches
+        them to the instance.
 
-    def __setstate__(self, state):
-        ctor_param, r = state
-        self.__init__(**ctor_param)
-        self._random_state = r
-        return self
+        Should be called after _construct_argparser, and during unpickling
+        (__setstate__)
```

Comment:
Does introducing this method change recommendations for subclassers? ISTM it's all hidden as long as `rv_generic.__init__` is called, but I'd rather ask to re-confirm.

Reply:
If subclassers call `rv_generic.__init__` (or if they subclass `rv_continuous`, then `rv_continuous.__init__`) everything should be ok. However, if they have special unpickling behaviour, overriding `__setstate__`, or expecting `__init__` to be called from `__setstate__`, then they may experience problems.